fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dto/ExtractedParameterValue.java

```diff
@@ -93,4 +80,99 @@ public String getName() {
     public void setName(String name) {
         this.name = name;
     }
+
+    /**
+     * @return the url
+     */
+    public String getUrl() {
+        return url;
+    }
+
+    /**
+     * @param url the url to set
+     */
+    public void setUrl(String url) {
+        this.url = url;
+    }
+
+    /**
+     * @return the version
+     */
+    public String getVersion() {
+        return version;
+    }
+
+    /**
+     * @param version the version to set
+     */
+    public void setVersion(String version) {
+        this.version = version;
+    }
+
+    @Override
+    public int compareTo(ExtractedParameterValue o) {
+        int retVal;
+        String thisClass = this.getClass().getName();
+        String otherClass = o.getClass().getName();
+        if (thisClass != null || otherClass != null) {
+            if (thisClass == null) {
+                return -1;
+            } else if (otherClass == null) {
+                return 1;
+            }
+            retVal = thisClass.compareTo(otherClass);
+            if (retVal != 0) {
+                return retVal;
+            }
+        }
+        String thisName = this.getName();
+        String otherName = o.getName();
+        if (thisName != null || otherName != null) {
+            if (thisName == null) {
+                return -1;
+            } else if (otherName == null) {
```

Comment:
names can never be null (because we need the name in order to store the value in the database. The null checks are therefore a little redundant. If we want to assert them as non-null, we should do so in the constructor.

Reply:
That is true.  Since this is a compareTo method, I just wanted to keep it generic and use the same logic for each field for consistency, and not have to care which fields are nullable, and which are not.